pkg/vpcmodel/grouping.go

```diff
@@ -438,37 +442,50 @@ func (g *GroupConnLines) groupInternalSrcOrDst(srcGrouping, groupVsi bool) {
 			}
 		}
 	}
-	g.GroupedLines = g.unifiedGroupedConnLines(res)
+	g.GroupedLines = unifiedGroupedConnLines(res, &g.cacheGrouped, false)
 }
 
-// Go over the grouping result and make sure all groups have a unified reference.
-// this is required due to the functionality treating self loops as don't cares - extendGroupingSelfLoops
+// Go over the grouping result and set groups s.t. all semantically equiv groups have a unified reference.
+// this is required for multivpc's context and at the end of the grouping in a single vpc context
+// the former is required since each vpc analysis and grouping is done separately
+// the latter is required due to the functionality treating self loops as don't cares - extendGroupingSelfLoops
 // in which both srcs and dsts are manipulated  but *GroupConnLines is not familiar
 // within the extendGroupingSelfLoops context and thus can not be done there smoothly
-func (g *GroupConnLines) unifiedGroupedConnLines(oldConnLines []*groupedConnLine) []*groupedConnLine {
+func unifiedGroupedConnLines(oldConnLines []*groupedConnLine, cacheGrouped *cacheGroupedElements,
```

Comment:
consider adding unit tests, demonstrating why this functionality is required for the described cases (both for multivpc's context and at the end of the grouping in a single vpc context)

Reply:
The effect of this unification is only in `drawio`  - the fixes originated in Haim's request. Its more natural to test it there. I can only test, or try to test, cumbersome side effects of the unification. 
Perhaps we can open an issue for this and see what it the best way to test it.

Reply:
ok, consider open a separate issue for this

Reply:
https://github.com/np-guard/vpc-network-config-analyzer/issues/429